R/fcm.R

```diff
@@ -221,7 +223,7 @@ fcm.tokens <- function(x, context = c("document", "window"),
         type <- types(x)
         boolean <- count == "boolean"
         temp <- qatd_cpp_fcm(x, length(type), weights, boolean, ordered)
-        temp <- as(temp, "dgCMatrix")
+        temp <- as(as(as(temp, "CsparseMatrix"), "generalMatrix"), "dMatrix")
```

Comment:
I think we can switch between dgT and dgC only single `as()`. We know that `qatd_cpp_fcm()` returns only dgT.

``` r
require(Matrix)
#> Loading required package: Matrix
sessionInfo()
#> R version 4.1.2 (2021-11-01)
#> Platform: x86_64-pc-linux-gnu (64-bit)
#> Running under: KDE neon User - Plasma 25th Anniversary Edition
#> other attached packages:
#> [1] Matrix_1.4-2
#> 
smat <- sparseMatrix(i = c(1, 3, 3), j = c(1, 1, 2), x = c(10, 20, 30), dims = c(3, 4))
class(smat)
#> [1] "dgCMatrix"
#> attr(,"package")
#> [1] "Matrix"
smat2 <- as(smat, "CsparseMatrix")
class(smat2)
#> [1] "dgCMatrix"
#> attr(,"package")
#> [1] "Matrix"
smat3 <- as(smat2, "TsparseMatrix")
class(smat3)
#> [1] "dgTMatrix"
#> attr(,"package")
#> [1] "Matrix"
```